src/yada/security.clj

```diff
@@ -315,8 +315,12 @@
   (when-not-cors-preflight ctx
 
     (let [authorization (call-fn-maybe (get-in ctx [:resource :authorization]) ctx)
-          realms (get-in ctx [:resource :access-control :realms])]
-
+          realms (get-in ctx [:resource :access-control :realms])
+          ;; We will return a 401 or 403 normally here.
+          ;; We could return a 404 to keep the resource hidden
+          ;; But we would still allow www-authenticate header in the error response
+          error-response (cond-> {:headers (select-keys (-> ctx :response :headers) ["www-authenticate"])}
```

Comment:
Perhaps this would be better broken out into a separate function? Also, when it's placed here, it is being calculated all the time, not just in error cases.